mods/pokebilities/abilities.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+exports.BattleAbilities = {
+	trace: {
+		inherit: true,
+		onUpdate: function (pokemon) {
+			if (!pokemon.isStarted) return;
+			let isAbility = this.effect.effectType === "Ability";
+			let possibleInnates = [];
+			let possibleTargets = [];
+			for (let i = 0; i < pokemon.side.foe.active.length; i++) {
+				let target = pokemon.side.foe.active[i];
+				if (target && !target.fainted) {
+					possibleInnates = possibleInnates.concat(target.innates);
+					possibleTargets = possibleTargets.concat(target.innates.map(innate => target));
```

Comment:
The question is why this isn't crashing because `target.innates` is possibly null?

Reply:
By which I mean, why isn't TypeScript catching this?

Reply:
It’s never defined as null, just string[] or undefined, and typescript didn’t catch this because this file is ignored in the tsconfig.

Reply:
Oh, I see.

@urkerab You're better than this :|

Reply:
Just for the record: In the original code, I know that `target.innates` will never be null because it always gets set for battles in that format. TypeScript doesn't know that, so it starts complaining. Unfortunately I ended up introducing a bug trying to quieten TypeScript, and as I didn't know about `dev-tools/globals.ts` I didn't allow TypeScript to tell me that.

Reply:
It's nice to use VS Code for this, because you can Cmd+click or Ctrl+click on a type, to go to the type definition.